### PR TITLE
Replace deprecated shell-parallel with concurrently

### PR DESCRIPTION
**Pull request by whymarrh**
This PR replaces the deprecated shell-parallel package with [concurrently](https://github.com/kimmobrunfeldt/concurrently)

### CODE REVIEW

File: test/e2e/run-web3.sh [hunk 1]
```diff
@@ -6,4 +6,8 @@ set -o pipefail
 
 export PATH="$PATH:./node_modules/.bin"
 
-shell-parallel -s 'static-server test/web3 --port 8080' -x 'sleep 5 && mocha test/e2e/web3.spec'
+concurrently --kill-others \
+  --names 'dapp,e2e' \
+  --prefix '[{time}][{name}]' \
+  'static-server test/web3 --port 8080' \
+  "bash -c 'sleep 5 && mocha test/e2e/web3.spec'"
```
Comment (Gudahtt) [inline test/e2e/run-web3.sh]: Why was the `bash -c` needed here?

Comment (whymarrh) [reply test/e2e/run-web3.sh]: It isn't, good catch

**Comment by Gudahtt** [comment]
Interesting; it looks like the e2e tests are failing because that `--kill-others` flag isn't managing to kill `ganache`. That's a shame.

**Comment by whymarrh** [comment]
Yeah it's a weird thing, possibly a case of https://github.com/kimmobrunfeldt/concurrently/issues/155 but I'm still debugging it

File: test/e2e/run-all.sh [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
 
-shell-parallel -s 'npm run ganache:start -- -b 2' -x 'sleep 5 && static-server test/e2e/contract-test --port 8080' -x 'sleep 5 && mocha test/e2e/metamask-ui.spec'
-shell-parallel -s 'npm run ganache:start -- -b 2' -x 'sleep 5 && static-server test/e2e/contract-test --port 8080' -x 'sleep 5 && mocha test/e2e/metamask-responsive-ui.spec'
-shell-parallel -s 'npm run ganache:start -- -d -b 2 --account=0x53CB0AB5226EEBF4D872113D98332C1555DC304443BEE1CF759D15798D3C55A9,25000000000000000000' \
-  -x 'sleep 5 && mocha test/e2e/from-import-ui.spec'
+concurrently --kill-others \
+  --names 'ganache,dapp,e2e' \
+  --prefix '[{time}][{name}]' \
+  --success first \
+  'npm run ganache:start -- -b 2 &> /dev/null' \
```
Comment (Gudahtt) [inline test/e2e/run-all.sh]: I'm not sure why, but this `&>` doesn't seem to be working. The `ganache` output is all printing to the CircleCI log.

Comment (whymarrh) [reply test/e2e/run-all.sh]: Yeah I noticed that too 🤔

**Comment by danjm** [comment]
@whymarrh to write wrapper script

File: test/e2e/run-all.sh [hunk 3]
```diff
@@ -1,12 +1,33 @@
 #!/usr/bin/env bash
 
+set -x
 set -e
 set -u
 set -o pipefail
 
 export PATH="$PATH:./node_modules/.bin"
+export GANACHE_ARGS='--quiet --blockTime 2'
```
Comment (Gudahtt) [inline test/e2e/run-all.sh]: The `--quiet` flag doesn't seem to be doing the trick either - the logs are still full of messages from `ganache`

Comment (whymarrh) [reply test/e2e/run-all.sh]: [](https://github.com/trufflesuite/ganache-cli/commit/e89d4325b94dd74e7f71a1adbc44d982c1b244d1)

Simply enough, the flag is newer than our version of the CLI. I'll update our CLI version.

**Review by Gudahtt — APPROVED** [review]
Looks good; there is still a bit more output from `ganache-cli` than there was before, but only at the beginning of the tests. That seems fine.

**Pull request merged**